desktop/core/src/desktop/middleware.py

```diff
@@ -953,3 +954,17 @@ def __call__(self, request):
       request.META['HTTP_X_FORWARDED_FOR'] = request.META['REMOTE_ADDR']
 
     return self.get_response(request)
+
+
+class CacheControlMiddleware:
+  def __init__(self, get_response):
+    self.get_response = get_response
+    self.custom_cache_control = CUSTOM_CACHE_CONTROL.get()
+
+  def __call__(self, request):
+    response = self.get_response(request)
+    if self.custom_cache_control:
+      response['Cache-Control'] = 'no-cache, no-store, must-revalidate'
```

Comment:
Curious: Maybe better to set the cache value consistent?
https://github.com/cloudera/hue/blob/876d88c95d03ca82cd7a95fcdd806720ec3fde74/desktop/conf.dist/hue.ini#L1629

Reply:
To prevent caching altogether, using the 'no-store' directive alone should be sufficient. The no-store directive instructs clients and intermediary caches (like proxies) not to store a cached copy of the response. In this case, eliminating max-age=0 and proxy-revalidate should not have any adverse effects, as the response will not be cached in the first place and 'no-cache' and 'must-revalidate' already enforce the desired caching behavior.

Reply:
Although its a very narrow use-case, but lets say `file_download_cache_control` will set that value for download operations but then when the response will flow form this middleware, the value will get overridden if the `custom_cache_control` is enabled, right?

Maybe we can make both values consistent or check if `Cache-Control` value is already set, then don't change it?
https://github.com/cloudera/hue/blob/c04065313c2765dd67bc9cbec02d36084fedc7fe/apps/filebrowser/src/filebrowser/views.py#L213